File: Distancia3P/Punto3d.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Distancia3P
{
    class Punto3d:Punto2d
    {
        protected decimal z;

        public decimal Z
        {
            get { return z; }
            set { z = value; }
        }

        public Punto3d() : base()
        {
            z = 0;
        }
        public Punto3d(decimal x, decimal y, decimal z) : base(x,y)
        {
            this.z = z;
        }

        public decimal Calcular_Distancia(Punto3d a)
        {
            return (decimal)(Math.Sqrt(Math.Pow((double)(x - a.X), 2) + Math.Pow((double)(y - a.Y), 2)+ Math.Pow((double)(z-a.Z),2)));

        }

        public override string ToString()
        {
            return x + " " + y + " " + z;
        }
    }
}

File: Distancia3PP/Punto3d.vb
Public Class Punto3d
    Inherits Punto2d

    Protected z As Decimal

    Public Property Z1() As Decimal
        Get
            Return z
        End Get
        Set(value As Decimal)
            z = value
        End Set
    End Property

    Public Sub New()
        MyBase.New()
        z = 0
    End Sub

    Public Sub New(x As Decimal, y As Decimal, z As Decimal)
        MyBase.New(x, y)
        Z1 = z
    End Sub

    Public Overloads Function Calcular_Distancia(a As Punto3d)
        Return Math.Sqrt(Math.Pow(x - a.X1, 2) + Math.Pow(y - a.Y1, 2) + Math.Pow(z - a.Z1, 2))
    End Function

    Public Overrides Function ToString() As String
        Return x + " " + y + " " + z
    End Function
End Class
